src/main/java/com/flightstats/hub/dao/aws/DynamoChannelConfigDaoLifecycle.java

```diff
@@ -0,0 +1,71 @@
+package com.flightstats.hub.dao.aws;
+
+import com.amazonaws.services.dynamodbv2.model.AttributeDefinition;
+import com.amazonaws.services.dynamodbv2.model.CreateTableRequest;
+import com.amazonaws.services.dynamodbv2.model.KeySchemaElement;
+import com.amazonaws.services.dynamodbv2.model.KeyType;
+import com.amazonaws.services.dynamodbv2.model.ProvisionedThroughput;
+import com.amazonaws.services.dynamodbv2.model.ScalarAttributeType;
+import com.flightstats.hub.config.AppProperties;
+import com.flightstats.hub.config.DynamoProperties;
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+public class DynamoChannelConfigDaoLifecycle extends AbstractIdleService {
+    private final DynamoUtils dynamoUtils;
+    private final AppProperties appProperties;
+    private final DynamoProperties dynamoProperties;
+
+    @Inject
+    public DynamoChannelConfigDaoLifecycle(DynamoUtils dynamoUtils, AppProperties appProperties, DynamoProperties dynamoProperties) {
+        this.dynamoUtils = dynamoUtils;
+        this.appProperties = appProperties;
+        this.dynamoProperties = dynamoProperties;
+    }
+
+    @Override
+    protected void startUp() {
+        initialize();
+    }
+
+    @Override
+    protected void shutDown() {
+    }
+
+    void initialize() {
+        String tableName = dynamoProperties.getChannelConfigTableName();
+        ProvisionedThroughput throughput = dynamoUtils.getProvisionedThroughput("channel");
+
+        if (!dynamoUtils.doesTableExist(tableName)) {
+            if (appProperties.isReadOnly()) {
+                String msg = String.format("Probably fatal error. Dynamo channel config table doesn't exist for r/o node.  %s", tableName);
+                log.error(msg);
+                throw new IllegalArgumentException(msg);
+            } else {
+                createTable(tableName, throughput);
+            }
+        } else if (!appProperties.isReadOnly()){
+            dynamoUtils.updateTable(tableName, throughput);
+        }
```

Comment:
FWIW, an if/elseif without a trailing else tends to make me uncomfortable. There's an implied set of conditions that allows a fall through and I have to re-figure out what that is when I read the code. I tend to add an else that logs the reason or at least a comment.
```
else {
    log.info("Read only hub and we validated the dynamoTable already exists.");
}
```
or
```
// else - Read only hub and we validated the dynamoTable already exists
```

Reply:
I think this might be more readable if the blocks were separated as readonly/not-readonly.  So something like:
```
if (isReadOnly) {
  if (tableExists) {
    throwTheError;
  }
} else {
  if (tableExists) {
    updateTable();
  } else {
    createTable();
  }
}
```

Reply:
Yep.  Fixed.